packages/flutter/lib/src/material/flexible_space_bar.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'constants.dart';
+import 'scaffold.dart';
+import 'theme.dart';
+
+class FlexibleSpaceBar extends StatefulComponent {
+  FlexibleSpaceBar({ Key key,this.title, this.image }) : super(key: key);
+
+  final Widget title;
+  final Widget image;
+
+  _FlexibleSpaceBarState createState() => new _FlexibleSpaceBarState();
+}
+
+class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
+
+  Widget build(BuildContext context) {
+    final double appBarHeight = Scaffold.of(context).appBarHeight;
```

Comment:
since this will crash without a scaffold, maybe have an assert block which looks for a scaffold and throws a WidgetError explaining the problem if it can't find one? See the pattern in material.dart.